src/components/Coupon/index.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { useState, useEffect } from "react";
+import { Typography, Card, Button } from "antd";
+import { useTranslation } from "react-i18next";
+
+import "./style.css";
+
+const { Title, Text } = Typography;
+
+const Coupon = (props) => {
+  const { t } = useTranslation();
+  const { id, createdAt, amounts, compName, img, isTesting } = props;
+  const [isSelected, setIsSelected] = useState({
+    price0: false,
+    price1: false,
+    price2: false,
+    price3: false,
+  });
+  const [selectedPrice, setSelectedPrice] = useState(0);
+  const [cart, setCart] = useState([]);
+  const handleSelect = (e) => {
+    setIsSelected({ [e.target.name]: true });
+    setSelectedPrice(parseInt(e.target.value, 10));
+  };
+  const handleAddToCart = () => {
+    const cartCopy = [...cart];
+    const isAdded = cartCopy.find(
+      (cartCoupon) => cartCoupon.id === id + selectedPrice
+    );
+    if (!isAdded && selectedPrice) {
+      const addedCoupn = {
+        id: id + selectedPrice,
```

Comment:
why are we adding the id with the selected price here ?

Reply:
because every price for a coupon is unique and should be sent separately to the cart, even though it's from the same restaurant.